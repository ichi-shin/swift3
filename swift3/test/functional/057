#!/bin/bash
# Copyright(c)2014 NTT corp. All Rights Reserved.
# DB1-1-2-2_9

. ./common

S3USER=tester

_s3_put /bucket -H "x-amz-acl:public-read"
_s3_get /bucket?acl -D - | _filter_curl xml

_s3_put /bucket?acl -H "x-amz-grant-write: emailAddress='xyz@amazon.com'" | _filter_curl xml
_s3_get /bucket?acl -D - | _filter_curl xml
