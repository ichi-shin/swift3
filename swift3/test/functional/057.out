QA output created by 057
> s3curl --id tester -- -X PUT -H x-amz-acl:public-read http://SWIFT_HOST/bucket... 200
> s3curl --id tester -- -X GET -D - http://SWIFT_HOST/bucket?acl... 200
HTTP/1.1 200 OK
Content-Length: LENGTH
Content-Type: text/html; charset=UTF-8
Date: DATE
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

<?xml version='1.0' encoding='UTF-8'?>
<AccessControlPolicy xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Owner>
    <ID>TESTER</ID>
    <DisplayName>TESTER</DisplayName>
  </Owner>
  <AccessControlList>
    <Grant>
      <Grantee xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="Group">
        <URI>http://acs.amazonaws.com/groups/global/AllUsers</URI>
      </Grantee>
      <Permission>READ</Permission>
    </Grant>
    <Grant>
      <Grantee xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="CanonicalUser">
        <ID>TESTER</ID>
        <DisplayName>TESTER</DisplayName>
      </Grantee>
      <Permission>FULL_CONTROL</Permission>
    </Grant>
  </AccessControlList>
</AccessControlPolicy>
> s3curl --id tester -- -X PUT -H x-amz-grant-write: emailAddress='xyz@amazon.com' http://SWIFT_HOST/bucket?acl... 501
<?xml version='1.0' encoding='UTF-8'?>
<Error>
  <Code>NotImplemented</Code>
  <Message>Not implemented.</Message>
  <RequestId>TXID</RequestId>
</Error>
> s3curl --id tester -- -X GET -D - http://SWIFT_HOST/bucket?acl... 200
HTTP/1.1 200 OK
Content-Length: LENGTH
Content-Type: text/html; charset=UTF-8
Date: DATE
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

<?xml version='1.0' encoding='UTF-8'?>
<AccessControlPolicy xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Owner>
    <ID>TESTER</ID>
    <DisplayName>TESTER</DisplayName>
  </Owner>
  <AccessControlList>
    <Grant>
      <Grantee xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="Group">
        <URI>http://acs.amazonaws.com/groups/global/AllUsers</URI>
      </Grantee>
      <Permission>READ</Permission>
    </Grant>
    <Grant>
      <Grantee xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="CanonicalUser">
        <ID>TESTER</ID>
        <DisplayName>TESTER</DisplayName>
      </Grantee>
      <Permission>FULL_CONTROL</Permission>
    </Grant>
  </AccessControlList>
</AccessControlPolicy>
